_python/main/views.py

```diff
@@ -38,6 +43,60 @@ def login_required_response(request):
         return redirect_to_login(request.build_absolute_uri())
 
 
+def hydrate_params(func):
```

Comment:
Interesting! I don't know if you saw: the Rails app also handles this at a higher level: https://github.com/harvard-lil/h2o/blob/develop/app/controllers/content/node_controller.rb#L6